functions/src/models/repository.model.ts

```diff
@@ -0,0 +1,45 @@
+// Dashboard hub firebase functions models/mappers
+import { DocumentData, DocumentReference, FirebaseAdmin, QuerySnapshot, WriteResult } from "../client/firebase-admin";
+
+export interface IRepository {
+  id: number;
+  fullName: string;
+  uid?: string;
+}
+
+export class RepositoryModel implements IRepository {
+  id: number;
+  fullName: string;
+  uid?: string;
+
+  public static getRepositoryReference(uid: string): DocumentReference {
+    return FirebaseAdmin.firestore().collection('repositories').doc(uid);
+  }
+
+  public static async getRepositoryById(id: number): Promise<DocumentData | null> {
```

Comment:
These looks more like `service` methods because the *model* knows the `id` already. I will look into refactoring this